stdlib/public/runtime/ProtocolConformance.cpp

```diff
@@ -169,6 +169,14 @@ static MetadataResponse getSuperclassForMaybeIncompleteMetadata(
   if (!classMetadata)
     return {_swift_class_getSuperclass(metadata), MetadataState::Complete};
 
+#if SWIFT_OBJC_INTEROP
+    // Artificial subclasses are not valid type metadata and
+    // tryGetCompleteMetadataNonblocking will crash on them. However, they're
+    // always fully set up, so we can just skip it and fetch the Subclass field.
+    if (classMetadata->isTypeMetadata() && classMetadata->isArtificialSubclass())
+      return {classMetadata->Superclass, MetadataState::Complete};
```

Comment:
In other places, we do this check for artificial subclasses in a loop, crawling up the superclass tree until we find a real Swift class. I don't know if that's just paranoia, though.

Reply:
That can happen at least in theory. No idea if it happens in reality, but it's good to handle. In this particular case, the loop is exterior to this function, as other code uses this function to walk the entire superclass chain. Returning artificial subclasses in that loop is unnecessary but harmless.